google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/migration/StagedArtifactNameConverterProvider.java

```diff
@@ -0,0 +1,124 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.migration;
+
+import static com.google.cloud.tools.intellij.appengine.cloud.AppEngineDeploymentConfiguration.STAGED_ARTIFACT_NAME_LEGACY;
+
+import com.google.cloud.tools.intellij.util.GctBundle;
+import com.intellij.conversion.CannotConvertException;
+import com.intellij.conversion.ConversionContext;
+import com.intellij.conversion.ConversionProcessor;
+import com.intellij.conversion.ConverterProvider;
+import com.intellij.conversion.ProjectConverter;
+import com.intellij.conversion.RunManagerSettings;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import org.jdom.Element;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A {@link ConverterProvider} for converting the staged artifact name in App Engine run
+ * configurations.
+ *
+ * <p>With the introduction of this field, existing run configurations may rely on the previous
+ * behavior for staging artifacts for custom runtime deployments. This behavior was to always rename
+ * the built artifact to "target.jar" or "target.war", depending on the type of artifact that was
+ * built.
+ *
+ * <p>This conversion flags all deployment configurations with a legacy bit that will be checked
+ * upon deployment. If set, the staged artifact name will be explicitly set to "target.jar" or
+ * "target.war" so the functionality is preserved.
+ */
+public final class StagedArtifactNameConverterProvider extends ConverterProvider {
+
+  public StagedArtifactNameConverterProvider() {
+    super("gcp-staged-artifact-name-converter");
+  }
+
+  @NotNull
+  @Override
+  public String getConversionDescription() {
+    return GctBundle.message("appengine.staged.artifact.name.converter.description");
+  }
+
+  @NotNull
+  @Override
+  public ProjectConverter createConverter(@NotNull ConversionContext context) {
+    return new ProjectConverter() {
+      @Override
+      public ConversionProcessor<RunManagerSettings> createRunConfigurationsConverter() {
+        return new StagedArtifactNameConversionProcessor(context);
+      }
+    };
+  }
+
+  @Override
+  public boolean canDetermineIfConversionAlreadyPerformedByProjectFiles() {
+    return false;
+  }
+
+  private static final class StagedArtifactNameConversionProcessor
+      extends ConversionProcessor<RunManagerSettings> {
+
+    static final Function<Element, Stream<Element>> TO_DEPLOYMENT_SETTINGS =
+        element -> {
+          Element deployment = element.getChild("deployment");
+          if (deployment != null) {
+            Element settings = deployment.getChild("settings");
+            if (settings != null) {
+              return Stream.of(settings);
+            }
+          }
+          return Stream.empty();
+        };
+
+    static final Predicate<Element> IS_GCP_APP_ENGINE_DEPLOY =
```

Comment:
wasn't there a suggestion to also filter for flex apps since isn't it that case that this will cause conversion on standard apps as is?

Reply:
Ah yes, let me add that restriction.

Reply:
Done, PTAL